src/xpk/core/nap.py

```diff
@@ -240,6 +240,29 @@ def create_autoprovisioning_config(
   return autoprovisioning_config, 0
 
 
+def get_cluster_metadata_configmap(args) -> tuple[dict, int]:
```

Comment:
What's the benefit of this function with respect to `get_cluster_configmap`?

Reply:
It reduces duplication in `get_autoprovisioning_node_selector_args` and `get_capacity_type_str_from_args_or_cluster_default`.  I don't have strong opinion here, happy to inline the func

Reply:
Maybe instead of inlining we could just use `get_cluster_configmap`? I would prefer not to add functions which main functionality is calling a different function with a const argument